scalyr_agent/builtin_monitors/url_monitor.py

```diff
@@ -87,12 +96,17 @@ def _initialize(self):
         # external change (e.g. misconfigured DNS server) could then prevent the agent from
         # starting up.
         self.url = self._config.get("url")
-        self.request_method = self._config.get("request_method") or "GET"
-        self.request_data = self._config.get("request_data") or None
-        self.request_headers = self._config.get("request_headers") or []
+        self.request_method = self._config.get("request_method")
+        self.request_data = self._config.get("request_data")
+        self.request_headers = self._config.get("request_headers")
         self.timeout = self._config.get("timeout")
         self.max_characters = self._config.get("max_characters")
         self.log_all_lines = self._config.get("log_all_lines")
+        if self.request_headers and type(self.request_headers) != JsonArray:
+            self._logger.emit_value(
```

Comment:
Don't log here since it is unclear if the log is even open.  Instead, `raise Exception("your reason for an error")`.  The code that is above this will log the exception in the proper place.  And we do want to kill the monitor if it looks like the configuration is not good.